slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -52,6 +52,40 @@ ::grpc::Status SLAMServiceImpl::GetPosition(ServerContext *context,
     return grpc::Status::OK;
 }
 
+::grpc::Status SLAMServiceImpl::GetPositionNew(
+    ServerContext *context, const GetPositionNewRequest *request,
+    GetPositionNewResponse *response) {
+    cartographer::transform::Rigid3d global_pose;
+    {
+        std::lock_guard<std::mutex> lk(viam_response_mutex);
```

Comment:
where is `lk` defined? 
I don't see it either here: https://cplusplus.com/reference/mutex/mutex/ or here: https://cplusplus.com/reference/mutex/lock_guard/

I assume it is the same as `lock` but I want to know where it is defined.

Reply:
Also, what unlocks the mutex?

Reply:
the mutex gets unlocked when we leave the scope the lock is defined for, so we create the lock, copy our values into a temp variable(here `global_pose`), then release the lock on `latest_global_pose` as we have exited its scope

Reply:
got it, the scope defines the duration of the lock.

Thanks for the explanation, still ramping up in C++!